src/main/java/org/assertj/core/internal/Objects.java

```diff
@@ -651,6 +653,23 @@ private boolean isActualIn(Object actual, Iterable<?> values) {
   }
 
   /**
+   * Assert that the given object is "deeply" equals to the other by comparing all fields recursively.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given object.
+   * @param other the object to compare {@code actual} to.
+   * @throws AssertionError if actual is {@code null}.
+   * @throws AssertionError if the other object is not an instance of the actual type.
+   * @throws AssertionError if the actual and the given object are not "deeply" equal.
+   */
+  public <A> void assertIsEqualToComparingFieldByFieldRecursively(AssertionInfo info, A actual, A other) {
+    assertNotNull(info, actual);
+    assertHasSameClassAs(info, actual, other);
+    if (!DeepEquals.deepEquals(actual, other))
+        throw failures.failure(info, shouldBeEqual(actual, other, StandardRepresentation.STANDARD_REPRESENTATION));
```

Comment:
No problem. :)

Time permitting I will try to add the missing features.

If you or somebody else comes up with a better solution meanwhile, I will close the pull request. I'm doing this to learn and to be able to use the functionality when testing. I'm not expecting to have ever pull request merged.